backend/src/main/java/org/schmeckis/backend/service/RezeptService.java

```diff
@@ -3,25 +3,80 @@
 import lombok.RequiredArgsConstructor;
 import org.schmeckis.backend.model.Rezept;
 import org.schmeckis.backend.model.dto.RequestRezept;
+import org.schmeckis.backend.model.submodel.Kategorie;
 import org.schmeckis.backend.repo.RezeptRepo;
+import org.schmeckis.backend.util.IdService;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Optional;
 
 @Service
 @RequiredArgsConstructor
 public class RezeptService {
 
-    RezeptRepo rezeptRepo;
+    private final RezeptRepo rezeptRepo;
 
-    public void createRezept(RequestRezept rezept) {
+    private final IdService idService;
+    public Rezept createRezept(RequestRezept rezept) {
         Optional<Rezept> isRezept = rezeptRepo.findByRezeptName(rezept.rezeptName());
         if (isRezept.isPresent()) {
             throw new IllegalArgumentException("Rezept already exists");
         }
-        rezeptRepo.save(new Rezept(rezept.rezeptName(), rezept.rezeptImageUrl(), rezept.rezeptKurzbeschreibung(), rezept.rezeptBeschreibung(), rezept.kategorieList()));
+        return rezeptRepo.save(new Rezept(idService.generateId(), rezept.rezeptName(), rezept.rezeptImageUrl(), rezept.rezeptKurzbeschreibung(), rezept.rezeptBeschreibung(), rezept.kategorieList()));
```

Comment:
du könntest dir auch eine .withId(String id) (oder sogar withNewRandomId()) Funktion zum Rezept Record hinzufügen. Dadurch wird es hier etwas leserlicher. 

Mit withNewRandomId() würdest du dann im Rezept Record die ID generieren. Damit könntest du den IdService eventuell hier aus dem Service entfernen.

Reply:
Gerade erste gesehen, dass du ein DTO (RequestRezept) verwendest. Sollte aber auch hier genauso gehen. 
Hier ein ähnliches Beispiel: https://github.com/kahlamari/PromptWhispers/blob/main/backend/src/main/java/in/kahl/promptwhispers/model/dto/PromptCreate.java